craft_application/commands/lifecycle.py

```diff
@@ -475,7 +486,13 @@ def _run(
             return
 
         emit.progress("Testing project")
-        self._services.get("testing").test(pathlib.Path.cwd())
+        testing_service.test(
+            pathlib.Path.cwd(),
+            tests=parsed_args.test_path,
+            shell=parsed_args.shell,
+            shell_after=parsed_args.shell_after,
```

Comment:
how do these options interact with shell, shell-after and debug for the packing itself? That is, if I run `testcraft test --shell-after` is it going to shell after packing, or after testing?

Reply:
interesting dilemma, if in the test step, shell and shell after should be after testing and hook into the spread shell, debug should enter a  shell related to the step.

We do not need to support this now as long as we have an issue to implement this

Reply:
`--debug` behaves as before, but also adds `-debug` to the spread command, entering a shell if a spread test fails.

I went back and checked and `--shell` and `--shell-after` were entering the shell as the `pack` command would (that is, during the `prime` step). I've now fixed it so it does what I intended, shelling in spread instead.

This is subtly different from the rest of our lifecycle commands, as spread's `-shell` and `-shell-after` engage per test being run rather than once per platform being packed, but I think that difference still matches user expectations.

Reply:
> I've now fixed it so it does what I intended, shelling in spread instead.

I think this is the bit that's troubling me, I'm not sure if this new behavior is desirable. To be clear: I think it's useful to have a way to shell/shell after/debug tests, I'm just not sure if we should reuse the regular lifecycle parameters for this. For instance, I can see how `testcraft test --debug` ending up inside the build instance because a lifecycle step failed could be surprising to users. And as you say, `-shell` (in spread) and `--shell` (the lifecycle param) behave subtly different.

I agree with Sergio, I think we should cook this a bit more

Reply:
What should this be reduced to then? Just `--debug`? Or perhaps even just adding the test paths?